e2e/rnTestUtilities.js

```diff
@@ -80,12 +80,111 @@ const assertAutotrackHierarchy = async (
   }
 };
 
-waitForEventsToFlush = async () => {
-  // Heap for iOS and Android flushes events every 15 seconds. Wait 16 seconds to ensure that
-  // all events are flushed to redis before asserting.
-  // :TODO:(jmtaber129): Make this wait shorter if/when we expose setting the
-  // flush frequency to the RN bridge.
-  await new Promise(resolve => setTimeout(resolve, 16000));
+const assertAndroidNavigationEvent = async (expectedPath, expectedType) => {
+  let props = {
+    path: {
+      string: expectedPath,
+    },
+  };
+  if (expectedType) {
+    props = {
+      ...props,
+      type: {
+        string: expectedType,
+      },
+    };
+  }
+
+  return assertAndroidEvent({
+    envId: '2084764307',
+    event: {
+      custom: {
+        name: 'reactNavigationScreenview',
+        properties: props,
+      },
+    },
+  });
+};
+
+const assertIosNavigationEvent = async (expectedPath, expectedType) => {
+  let props = ['path', expectedPath];
+  if (expectedType) {
+    props = [...props, 'type', expectedType];
+  }
+  return assertIosPixel({
+    t: 'reactNavigationScreenview',
+    k: props,
+  });
+};
+
+const assertNavigationEvent = async (expectedPath, expectedType) => {
+  if (device.getPlatform() === 'android') {
+    return assertAndroidNavigationEvent(expectedPath, expectedType);
+  } else if (device.getPlatform() === 'ios') {
+    return assertIosNavigationEvent(expectedPath, expectedType);
+  } else {
+    throw new Error(`Unknown device type: ${device.getPlatform()}`);
+  }
+};
+
+pollForSentinel = async (sentinelValue, timeout = 60000) => {
+  console.log(
+    `--- Waiting for ${sentinelValue} sentinel event.  This will timeout in 60s ---`
+  );
+  const startTick = Date.now();
+
+  // Give it a few seconds at first.
+  await new Promise(resolve => setTimeout(resolve, 3000));
+
+  while (Date.now() - startTick <= timeout) {
+    const eventName = `${sentinelValue.toUpperCase()}_SENTINEL`;
+    if (device.getPlatform() === 'ios') {
+      const event = {
+        a: '2084764307',
```

Comment:
Seems unrelated to PR, but could probably extract this to some ALL_CAPS const and reuse it.

Reply:
Yeah, totally... done...